Sources/TensorFlow/Operators/NN.swift

```diff
@@ -185,7 +185,7 @@ func _vjpConv2DBackpropInput<Scalar: TensorFlowFloatingPoint>(
     let value = conv2DBackpropInput(x, shape: shape, filter: filter,
                                     strides: strides, padding: padding, dilations: dilations)
     return (value, { v in
-        (conv2DBackpropFilter(x, input: v, filterSizes: shape, strides: strides,
+        (conv2DBackpropFilter(x, input: v, filterSizes: filter.shapeTensor, strides: strides,
```

Comment:
Using `filter.shapeTensor` here is less than ideal because the `filter` value would be captured in the backward pass, while we only need its `shapeTensor`. Instead, you can explicitly capture `filter.shapeTensor` as a separate value (`filterShape`, in my suggestion above), and use `filterShape` here directly.
```suggestion
        (conv2DBackpropFilter(x, input: v, filterSizes: filterShape, strides: strides,
```